docs/cookbook.md

```diff
@@ -0,0 +1,412 @@
+    }
+
+    public function setName($name): void
+    {
+        $this->name = $name;
+    }
+}
+```
+
+Then, create the doctrine configuration for this entity to declare the entity fields and table name where the entity will be persisted.
+
+```yaml
+#Acme\Bundle\SupplierBundle\Resources\config\doctrine\Supplier.orm.yml
+Acme\Bundle\SupplierBundle\Entity\Supplier:
+    type: entity
+    table: refdata_supplier
+    repositoryClass: Pim\Bundle\CustomEntityBundle\Entity\Repository\CustomEntityRepository
+    fields:
+        id:
+            type: integer
+            id: true
+            generator:
+                strategy: AUTO
+        code:
+            type: string
+            length: 25
+            nullable: false
+            unique: true
+        name:
+            type: string
+            length: 255
+            nullable: false
+```
+
+## Update the database
+
+Each new reference data is persisted in a dedicated table. You can see what will be the create table SQL query by executing this command :
+```bash
+bin/console doctrine:schema:update --dump-sql
+```
+
+In our case, this will output something like this : 
+```sql
+CREATE TABLE refdata_supplier (id INT AUTO_INCREMENT NOT NULL, code VARCHAR(25) NOT NULL, name VARCHAR(255) NOT NULL, UNIQUE INDEX UNIQ_EF5B01F877153098 (code), PRIMARY KEY(id));
+```
+
+Now run the command by replacing the `--dump-sql` option by `--force` to create the table in the database :
+```bash
+bin/console doctrine:schema:update --force
+```
+
+You can check that the table is correctly created in your database.
+ 
+## Your reference data as an attribute
+Now declare the new reference data in the application : 
+```yaml
+#app/config/config.yml
+pim_reference_data:
+    supplier:
+        class: Acme\Bundle\SupplierBundle\Entity\Supplier
+        type: simple
+```
+
+To check every thing is OK so far, execute the following commands :
+```bash
+#check if the entity doctrine configuration is OK
+bin/console doctrine:mapping:info
+```
+```bash
+#check if the reference data is correct
+bin/console pim:reference-data:check
+```
+
+At this point, the reference data can already be used as an attribute in your families. Try to create a new "reference data simple select" attribute.
+
+## Reference data CRUD
+
+### Menu
+A new menu item can be added to access the reference data list. To achieve that, declare the menu item by creating a new configuration file :
+```yaml
+#Acme\Bundle/SupplierBundle/Resources/config/form_extensions/menu.yml
+extensions:
+    acme-menu-supplier:
+        module: pim/menu/item
+        parent: pim-menu-reference_data-navigation-block
+        position: 10
+        config:
+            title: acme.menu.item.reference_data.supplier
+            to: pim_customentity_index
+            routeParams:
+                customEntityName: supplier
+```
+At this point the item menu should be visible at the end of the menu. You may have to clear your cache (including the front end cache) to see it.
```

Comment:
If you have a link to the official documentation or GitHub repository for that.

Reply:
I didn't find any documentation for that. I did found that : https://docs.akeneo.com/2.3/maintain_pim/first_aid_kit/index.html#step-8-did-you-clear-the-cache but it's not complete, so it's not usable.